### PR TITLE
SW-4503 Implement v2 enhanced top bar with select all, select none, enhanced selection text

### PR DESCRIPTION
**Pull request by nickgraz**
V2 wrapper that changes functionality if the `enhancedTopBarSelectionConfig` is there or not. The wrapper will decorate the config with the necessary functionality to render the text without passing more data than necessary into the toolbar component.

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/web-components/450?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #449** <a href="https://app.graphite.dev/github/pr/terraware/web-components/449?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #450** <a href="https://app.graphite.dev/github/pr/terraware/web-components/450?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/web-components/450?utm_source=stack-comment).

File: src/components/table/EnhancedTableToolbar.tsx [hunk 1]
```diff
@@ -18,19 +19,41 @@ const styles = makeStyles((theme: Theme) => ({
...
+      return renderNumSelectedText(numSelected);
+    }
+
+    const { handleSelectAll, handleSelectNone, renderSelectAllText, renderSelectNoneText, renderEnhancedNumSelectedText } = topBarSelectionConfig;
+
+    return (
+      <>
+        {renderEnhancedNumSelectedText ? renderEnhancedNumSelectedText() : renderNumSelectedText(numSelected)}{' '}
+        {!isAllSelected && handleSelectAll && renderSelectAllText && <Link onClick={handleSelectAll}>{renderSelectAllText()}</Link>}
+        {isAllSelected && handleSelectNone && renderSelectNoneText && <Link onClick={handleSelectNone}>{renderSelectNoneText()}</Link>}
```
Comment (nickgraz) [inline src/components/table/EnhancedTableToolbar.tsx:48]: The parent will have to implement the styling

**Comment by nickgraz** [comment]
publish rc

**Comment by nickgraz** [comment]
publish rc

**Review by karthikbtf — APPROVED** [review]
nice
just the vercel deployment issue outstanding

**Comment by nickgraz** [comment]
Yea the issue seems to be related to us using node 16 locally, but vercel is using node 18.

https://github.com/terraware/web-components?tab=readme-ov-file#common-issues





I can try to enable legacy OpenSSL provider when I get back, but we should probably upgrade to node 18. Everything works fine in the produced npm package.

**Comment by nickgraz** [comment]
Seems to be unrelated to my work, 
see this pr - https://github.com/terraware/web-components/pull/451

**Comment by nickgraz** [comment]
Gonna set the node version in vercel to 16 and see if that works, that gets EOLed some time soon though.

**Comment by nickgraz** [comment]
June 14

**Pull request merged**